tests/foreman/ui/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,250 @@
+
+    nw_id = module_azurerm_cr.available_networks()['results'][-1]['id']
+    return entities.ComputeAttribute(
+        compute_profile=COMPUTE_PROFILE_SMALL,
+        compute_resource=module_azurerm_cr,
+        vm_attrs={
+            "resource_group": AZURERM_RG_DEFAULT,
+            "vm_size": AZURERM_VM_SIZE_DEFAULT,
+            "username": module_azurerm_finishimg.username,
+            "password": settings.azurerm.password,
+            "platform": AZURERM_PLATFORM_DEFAULT,
+            "script_command": "touch /var/tmp/text.txt",
+            "script_uris": AZURERM_FILE_URI,
+            "interfaces_attributes": {
+                "0": {"public_ip": "Static", "private_ip": "false", "network": nw_id}
+            },
+        },
+    ).create()
+
+
+@pytest.fixture(scope='module')
+def module_azure_hg(
+    module_azurerm_cr,
+    module_azure_cp_attrs,
+    module_architecture,
+    module_os,
+    module_puppet_environment,
+    module_smart_proxy,
+    module_domain,
+    module_loc,
+    module_org,
+):
+    """ Create hostgroup """
+
+    return entities.HostGroup(
+        architecture=module_architecture,
+        compute_resource=module_azurerm_cr,
+        compute_profile=COMPUTE_PROFILE_SMALL,
+        domain=module_domain,
+        location=[module_loc],
+        environment=module_puppet_environment,
+        puppet_proxy=module_smart_proxy,
+        puppet_ca_proxy=module_smart_proxy,
+        content_source=module_smart_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+    ).create()
+
+
+@tier4
+def test_positive_end_to_end_azurerm_ft_host_provision(
+    session,
+    azurermclient,
+    module_azurerm_finishimg,
+    module_azurerm_cr,
+    module_domain,
+    module_org,
+    module_loc,
+    module_azure_hg,
+):
+
+    """Provision Host with hostgroup and Compute-profile using
+    finish template on AzureRm compute resource
+
+    :id: d64d249d-70a2-4329-bff4-3b50b8596c44
+
+    :expectedresults:
+            1. Host is provisioned.
+            2. Host is deleted Successfully.
+
+    :CaseLevel: System
+    """
+
+    hostname = gen_string('alpha')
+    fqdn = '{}.{}'.format(hostname, module_domain.name).lower()
+
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision Host
+        try:
+            skip_yum_update_during_provisioning(template='Kickstart default finish')
+            session.host.create(
+                {
+                    'host.name': hostname,
+                    'host.hostgroup': module_azure_hg.name,
+                    'provider_content.operating_system.root_password': gen_string('alpha'),
+                    'provider_content.operating_system.image': module_azurerm_finishimg.name,
+                }
+            )
+
+            host_info = session.host.get_details(fqdn)
+            assert session.host.search(fqdn)[0]['Name'] == fqdn
```

Comment:
`host.get_details()` uses `.search()` during `step`, so this assertion is redundant - it must pass if `get_details()` succeeded.

Reply:
good catch , removed that assertion as not required and make sense much here.